account_customer_wallet/tests/common.py

```diff
@@ -0,0 +1,124 @@
+            "account_customer_wallet.account_journal_customer_wallet_demo"
+        )
+        cls.sale_journal = cls.env["account.journal"].search(
+            [("type", "=", "sale")], limit=1
+        )
+        cls.payment_method = cls.env.ref("account.account_payment_method_manual_in")
+        cls.cash_account = cls.env["account.account"].search(
+            [("account_type", "=", "asset_cash")], limit=1
+        )
+        cls.company_id = cls.env.company
+
+    def _create_move(self, debit=0, credit=0, partner=None):
+        if partner is None:
+            partner = self.partner
+
+        self.env["account.move"].create(
+            {
+                "journal_id": self.customer_wallet_journal.id,
+                "line_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "debit": debit,
+                            "credit": credit,
+                            "partner_id": partner.id,
+                            "account_id": self.customer_wallet_account.id,
+                        },
+                    ),
+                    (
+                        0,
+                        0,
+                        {
+                            "debit": credit,
+                            "credit": debit,
+                            "partner_id": partner.id,
+                            "account_id": self.cash_account.id,
+                        },
+                    ),
+                ],
+            }
+        )
+
+    def _create_sale_invoice(self, invoice_type, amount, partner=None):
+        if partner is None:
+            partner = self.partner
+        invoice = self.env["account.move"].create(
+            {
+                "journal_id": self.sale_journal.id,
+                "move_type": invoice_type,
+                "partner_id": partner.id,
+                "line_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "name": "Invoice line",
+                            "quantity": 1,
+                            "price_unit": amount,
+                            "product_id": self.sale_product.id,
+                            "account_id": self.sale_account.id,
+                        },
+                    )
+                ],
+            }
+        )
+        invoice.action_post()
+        return invoice
+
+    def _create_payment(self, invoice, amount=0, partner=None):
+        if partner is None:
+            partner = self.partner
+        register = (
+            self.env["account.payment.register"]
+            .with_context(
+                active_model="account.move",
+                active_ids=invoice.ids,
+            )
+            .create(
+                {
+                    "journal_id": self.customer_wallet_journal.id,
+                }
+            )
+        )
+        return register._create_payments()
```

Comment:
So this kind of works, and triggers the balance check in `account_payment.py` correctly, but it does NOT create `account.move.line`s that affect the customer account wallet. I do not know which code to alter where to create an invoice OR a payment register that credits the wallet account.

Reply:
I just tested.
In V16, create a payment, by default, create an account move line with the account defined globally on the company in the field "Outstanding Receipts Account".
1) I don't know why.
2) I don't know how to handle correctly the payment in V16.

Reply:
If we set the wallet account on the journal on that field, it works.

![image](https://user-images.githubusercontent.com/3407482/233695669-d682e4bd-47f9-4b3f-9372-970c153b9a1f.png)